### ISSUE TITLE
Example Upload timeline CSV fails with No columns to parse from file

### ISSUE THREAD

**Issue by knodesec** [google/timesketch#1666]
## Overview
This is day one with using TimeSketch, so hopefully I'm not doing something obviously stupid..  

Example CSV import fails with "pandas.errors.EmptyDataError: No columns to parse from file"

Whilst trying to import a CSV file that I had generated from parsing an SQL log, I was encountering error related to the key "datetime". I decided to step it back to the most minimal scenario to step through it and find out where the problem was. However, when trying the most basic import, I still couldn't get it to work. I created the following file directly from the example [README](https://github.com/google/timesketch/blob/master/docs/CreateTimelineFromJSONorCSV.md):

```csv
message,timestamp,datetime,timestamp_desc,extra_field_1,extra_field_2
A message,1331698658276340,2015-07-24T19:01:01+00:00,Write time,foo,bar
```

When using the web UI to import the file, I get the following error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/timesketch/lib/tasks.py", line 647, in run_csv_jsonl
    for event in read_and_validate(file_handle):
  File "/usr/local/lib/python3.8/dist-packages/timesketch/lib/utils.py", line 154, in read_and_validate_csv
    reader = pandas.read_csv(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 686, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 452, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 946, in __init__
    self._make_engine(self.engine)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 1178, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 2008, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 540, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
```

The info box in the UI for the file shows:

Imported | Provider | Context | User | File on disk | File size | Original filename | Data label
-- | -- | -- | -- | -- | -- | -- | --
2021-03-04 19:13:29 | WebUpload | timesketch-csv-test-example.csv | knodesec | /usr/share/timesketch/upload/7be2fe9d7daa45529974327120657cc2 | 144B | timesketch-csv-test-example.csv

and checking the file on the worker container

```
docker exec -it 240b /bin/bash
root@240b2cc41450:/usr/share/timesketch/upload# cat -A 7be2fe9d7daa45529974327120657cc2
message,timestamp,datetime,timestamp_desc,extra_field_1,extra_field_2^M$
A message,1331698658276340,2015-07-24T19:01:01+00:00,Write time,foo,bar^M$
```
I don't see anything interesting in the docker-compose logs, however the timesketch worker logs show the following:
```
[2021-03-04 19:11:38,462] celery.app.trace/INFO Task timesketch.lib.tasks.run_csv_jsonl[45b19bce-64e4-4fab-8750-88a802dc35b8] succeeded in 0.3282905069645494s: None
[2021-03-04 19:13:30,001] celery.worker.strategy/INFO Received task: timesketch.lib.tasks.run_csv_jsonl[a2e76351-3c39-4887-bfa9-1a511e479a47]
[2021-03-04 19:13:30,001] celery.pool/DEBUG TaskPool: Apply <function _fast_trace_task at 0x7ff6e1058040> (args:('timesketch.lib.tasks.run_csv_jsonl', 'a2e76351-3c39-4887-bfa9-1a511e479a47', {'lang': 'py', 'task': 'timesketch.lib.tasks.run_csv_jsonl', 'id'
: 'a2e76351-3c39-4887-bfa9-1a511e479a47', 'shadow': None, 'eta': None, 'expires': None, 'group': None, 'retries': 0, 'timelimit': [None, None], 'root_id': 'a2e76351-3c39-4887-bfa9-1a511e479a47', 'parent_id': None, 'argsrepr': "('/usr/share/timesketch/uploa
d/7be2fe9d7daa45529974327120657cc2', '', 'timesketch-csv-test-example', '43ee01b2286b41eeb9959d74caaca926', 'csv', 16)", 'kwargsrepr': '{}', 'origin': 'gen9@240b2cc41450', 'reply_to': '486dc1e3-52d9-36c1-8f88-6c82f498d3f5', 'correlation_id': 'a2e76351-3c39
-4887-bfa9-1a511e479a47', 'hostname': 'celery@4e2249d1eab1', 'delivery_info': {'exchange': '', 'routing_key': 'celery', 'priority': 0, 'redelivered': None}, 'args': ['/usr/share/timesketch/upload/7be2fe9d7daa45529974327120657cc2', '', 'timesketch-csv-test-
example', '43ee01b2286b41eeb9959d74caaca926', 'csv', 16], 'kwargs': {}},... kwargs:{})
[2021-03-04 19:13:30,002] celery.worker.request/DEBUG Task accepted: timesketch.lib.tasks.run_csv_jsonl[a2e76351-3c39-4887-bfa9-1a511e479a47] pid:22
[2021-03-04 19:13:30,002] timesketch.tasks/INFO Index timeline [timesketch-csv-test-example] to index [43ee01b2286b41eeb9959d74caaca926] (source: csv)
[2021-03-04 19:13:30,003] urllib3.connectionpool/DEBUG Starting new HTTP connection (1): elasticsearch:9200
[2021-03-04 19:13:30,005] urllib3.connectionpool/DEBUG http://elasticsearch:9200 "GET / HTTP/1.1" 200 538
[2021-03-04 19:13:30,007] urllib3.connectionpool/DEBUG http://elasticsearch:9200 "HEAD /43ee01b2286b41eeb9959d74caaca926 HTTP/1.1" 404 0
[2021-03-04 19:13:30,158] urllib3.connectionpool/DEBUG http://elasticsearch:9200 "PUT /43ee01b2286b41eeb9959d74caaca926 HTTP/1.1" 200 91
[2021-03-04 19:13:30,177] timesketch.tasks/ERROR Error: No columns to parse from file
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/timesketch/lib/tasks.py", line 647, in run_csv_jsonl
    for event in read_and_validate(file_handle):
  File "/usr/local/lib/python3.8/dist-packages/timesketch/lib/utils.py", line 154, in read_and_validate_csv
    reader = pandas.read_csv(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 686, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 452, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 946, in __init__
    self._make_engine(self.engine)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 1178, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/parsers.py", line 2008, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 540, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
```

I have also tried this with CRLF vs LF line endings but it doesn't appear to make a difference, all uploads fail with the same error.
```
root@240b2cc41450:/usr/share/timesketch/upload# cat -A cc13cfa4ffc0472b8af13c640c73e687
message,timestamp,datetime,timestamp_desc,extra_field_1,extra_field_2$
A message,1331698658276340,2015-07-24T19:01:01+00:00,Write time,foo,bar$
A message,1331698658276340,2015-07-24T19:01:01+00:00,Write time,foo,bar$
A message,1331698658276340,2015-07-24T19:01:01+00:00,Write time,foo,bar$
A message,1331698658276340,2015-07-24T19:01:01+00:00,Write time,foo,bar$
root@240b2cc41450:/usr/share/timesketch/upload# cat -A 7be2fe9d7daa45529974327120657cc2
message,timestamp,datetime,timestamp_desc,extra_field_1,extra_field_2^M$
A message,1331698658276340,2015-07-24T19:01:01+00:00,Write time,foo,bar^M$
```

## Version / Installation

Timesketch version: 20210224

Installed via the [README](https://github.com/google/timesketch/blob/master/docs/Installation.md)
Here is a snippet of the resulting `config.env` 
```
# Timesketch version to run. Latest is build from the master branch and a release
# number is build from a release tag. Using latest means that you are running
# the bleeding edge version and we cannot guarantee that it will not be broken.
TIMESKETCH_VERSION=latest

# Number of web workers to run. Rule: (num cores * 2) + 1
NUM_WSGI_WORKERS=4

# Log level for import and analysis workers.
WORKER_LOG_LEVEL=debug

# Version of Elasticsearch to run
ELASTIC_VERSION=7.10.2
```

**Comment by kiddinn** [comment]
I will take a look at this, we recently changed the backend ingestor to use pandas (#1534) and I expected there to be some issues that could come up with that.

You can also go back to the latest released version and this should work, but thank you for the report, I'll see if I can reproduce this and then fix it.

**Comment by kiddinn** [comment]
ok, so I tested this with the simple file that you provided.

This is easily reproduced and fails with the web UI uploads, but succeeds without issues using the importer client. 

![image](https://user-images.githubusercontent.com/8701947/110327729-40365000-8012-11eb-8f15-c4d325a737a4.png)

**Comment by kiddinn** [comment]
this should be fixed in #1667 1667
